### ISSUE TITLE
Minimal set of translates of a shifted monoid

### ISSUE THREAD

**Issue by thchr** [Normaliz/Normaliz#433]
I was hoping to be able to extract the following via Normaliz, but have been unable to do figure out how from the manual.

What I want to do is the following:
Suppose I have a homogeneous cone *C* defined by a coefficient matrix **A**, s.t. $C$ is spanned by **Ax** with **x**  ∈ ℝ<sub>≥0</sub>. There is an associated monoid *M* = ℤ<sup>*d*</sup> ∪ *C*. I can compute an associated Hilbert basis {**h**<sub>*i</sub> via Normaliz:
```py
C = Cone(inequalities = A)
C.Compute("HilbertBasis")
```

Now, what I'm actually interested in is not *M*, but *M*′ = ℤ<sup>*d*</sup> ∪ *C*′, which is obtained from a translated cone *C*′, i.e., generated by **Ax**-**b** for some **b** and again with **x**  ∈ ℝ<sub>≥0</sub>.
It seems to me that it should be possible to express *M*′ as a union of a finite set of elements, say, {**z**}, with *M* such that
*M*′ = ∪<sub>**z** ∈ {**z**}</sub> **z**+M. I am not exactly sure what this set is called - but maybe something like "minimal set of translates" or "generating set of translates" of *M*′ in *M*. Equivalently, all elements of *M*′ would then be expressible as a non-negative combination of Hilbert basis vectors from *M* and some element from this set {**z**} of translates.

From the manual, I believe I understood that this situation could be set up by specifying `Cone(inhom_inequalities = Ab)`, with `Ab` denoting the concatenation of **A** and **b**, but I can then not figure out how to proceed from there. Maybe it's already in the manual - but lacking the full vocabulary of polytope and monoid theory, I cannot identify it.

Any help would be greatly appreciated.

**Comment by w-bruns** [comment]
To be on the safe side: do you mean that M is the **intersection** of the cone and the lattice? Etc.

**Comment by thchr** [comment]
Ah, yes - all my cups should have been caps; intersection, not unions.

In the meantime, after correcting an example I was using to experiment with, I am wondering if the functionality I was looking for might simply be `C.LatticePoints()`?

**Comment by w-bruns** [comment]
Unfortunately it will take me more time than I thought. Please try nevertheless.

**Comment by thchr** [comment]
I have tried, and it seems indeed to work exactly as I had hoped it would - it's wonderful. I've compared with more "pedestrian" approaches for a handful of settings that I know, and things work out there as they ought.

The only remaining question I have is what the difference is between `C.LatticePoints()` and `C.ModuleGenerators()`? In my tests so far, they give the same answers.

Many thanks!

**Comment by w-bruns** [comment]
I was not sure about LastticePpoints. I will check again. If it works for you, very good.

**Comment by w-bruns** [comment]
Indeed, ModuleGenerators and LtticePoints give the same answer, at least for inmhomogeneous input.

**Comment by thchr** [comment]
> Borrowing terminology from rings and modules, we can say that M' is a finitely generated module over the monoid M of lattice points in C.

Sorry to keep pestering you; one final question (feel free to ignore): In the this terminology (also, should it be "of lattice points in C'" rather than C?"), why do we call the associated structure a module? I thought a module required closure under a multiplication ring, which would seem to absent here.

**Comment by w-bruns** [comment]
Sems to have been clarified.
